### PR TITLE
Include the namespace when detecting infoboxes for Builder

### PR DESCRIPTION
**Pull request by sebastianmarzjan**
Introduce an explicit check against namespace when determining Infobox Builder support to avoid false positives

@idradm

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php [hunk 1]
```diff
@@ -9,20 +9,15 @@ class PortableInfoboxBuilderHelper {
...
 	public static function isInfoboxTemplate( $title ) {
-		$tc = new TemplateClassificationService();
-		$isInfobox = false;
-
-		try {
-			$type = $tc->getType( F::app()->wg->CityId, $title->getArticleID() );
-			$isInfobox = ( $type === TemplateClassificationService::TEMPLATE_INFOBOX );
-		} catch ( Swagger\Client\ApiException $e ) {
-			// If we cannot reach the service assume the default (false) to avoid overwriting data
+		if ( $title->getNamespace() === NS_TEMPLATE ) {
```
Comment (idradm) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php]: You could use `inNamespace()` method available in title object, and make it a single return statement.

Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php]: Nice suggestion, done in 49c5a9d

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php [hunk 2]
```diff
@@ -50,16 +41,17 @@ public static function getUrlPath( $titleText ) {
...
 	 */
 	public static function canUseInfoboxBuilder( $title, $user ) {
+		$infoboxes = \PortableInfoboxDataService::newFromTitle( $title )->getInfoboxes();
+
 		return PortableInfoboxBuilderHelper::isInfoboxTemplate( $title )
-			&& ( new \PortableInfoboxBuilderService() )->isValidInfoboxArray(
-				\PortableInfoboxDataService::newFromTitle( $title )->getInfoboxes()
-			)
-			&& ( new \Wikia\TemplateClassification\Permissions() )->userCanChangeType( $user, $title );
+			   && ( $title->isKnown() && !empty( $infoboxes ) )
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php:52]: what is the case that isKnown() covers for us that exists() doesn't?

**Comment by sebastianmarzjan** [comment]
One question, the changes starting at 63228e3 look good to me

**Comment by dianafa** [comment]
LG

File: extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php (hunk 2 above)
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/services/PortableInfoboxBuilderHelper.class.php:52]: OK, no difference in our case

**Pull request merged**